sunpy/map/maputils.py

```diff
@@ -80,7 +80,22 @@ def map_edges(smap):
     return top, bottom, left_hand_side, right_hand_side
 
 
-@u.quantity_input
+def _is_coordinate_helioprojective(coordinates):
+    """
+    Checks if the coordinate is Helioprojective
```

Comment:
```suggestion
def _verify_coordinate_helioprojective(coordinates):
    """
    Raises an error if the coordinate is not in the
    `~sunpy.coordinates.frames.Helioprojective` frame.
```
Change this private function to raise the error if needed